packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  buyAudioSelectors,
+  BuyAudioStage,
+  formatNumberString
+} from '@audius/common'
+import { IconCaretDown } from '@audius/stems'
+import { useSelector } from 'react-redux'
+
+import LoadingSpinner from 'components/loading-spinner/LoadingSpinner'
+import { ToggleCollapseButton } from 'components/toggle-collapse-button'
+
+import { IconAUDIO, IconSOL, IconUSD } from './Icons'
+import styles from './InProgressPage.module.css'
+
+const { getAudioPurchaseInfo, getBuyAudioFlowStage } = buyAudioSelectors
+
+const messages = {
+  pleaseHold: 'Please hold on. This may take a few moments.',
+  completeWithCoinbase: 'Complete this step with Coinbase',
+  step1: 'Step 1',
+  step2: 'Step 2',
+  purchasingSol: 'Purchasing SOL',
+  convertingSol: 'Converting SOL to $AUDIO',
+  moreInfo: 'More Info',
+  lessInfo: 'Less Info',
+  usd: 'USD',
+  sol: 'SOL',
+  audio: '$AUDIO'
+}
+
+export const InProgressPage = () => {
+  const purchaseInfo = useSelector(getAudioPurchaseInfo)
+  const buyAudioFlowStage = useSelector(getBuyAudioFlowStage)
+  const isStepOne = buyAudioFlowStage === BuyAudioStage.PURCHASING
+  let firstToken, secondToken
+  if (purchaseInfo?.isError === false) {
+    const solToken = {
+      label: messages.sol,
+      icon: <IconSOL />,
+      amount: formatNumberString(purchaseInfo.estimatedSOL.uiAmountString, {
+        maxDecimals: 2
+      })
+    }
+    if (isStepOne) {
+      firstToken = {
+        label: messages.usd,
+        icon: <IconUSD />,
+        amount: formatNumberString(purchaseInfo.estimatedUSD.uiAmountString, {
+          minDecimals: 2,
+          maxDecimals: 2
+        })
+      }
+      secondToken = solToken
+    } else {
+      firstToken = solToken
+      secondToken = {
+        label: messages.audio,
+        icon: <IconAUDIO />,
+        amount: formatNumberString(
+          purchaseInfo.desiredAudioAmount.uiAmountString,
+          {
+            maxDecimals: 2
+          }
+        )
+      }
+    }
+  }
+  return (
+    <div className={styles.inProgressPage}>
+      <div className={styles.header}>{messages.pleaseHold}</div>
+      <div className={styles.loader}>
+        <LoadingSpinner className={styles.spinner} />
+        <span className={styles.headerCaps}>
+          {isStepOne ? messages.step1 : messages.step2}
+        </span>
+      </div>
+      {isStepOne ? (
+        <div className={styles.callToAction}>
+          {messages.completeWithCoinbase}
+        </div>
+      ) : null}
+      <ToggleCollapseButton
+        id='buy-audio-more-info'
```

Comment:
so i think the button itself shouldn't have the id, but it should "aria-controls" something with that id?

Reply:
haven't looked at implementation too closely, but i might do: 
```
<CollapsibleSection aria-controls="content-id"><div id="content-id" .../> </CollapsibleSection />
```

Prefer that naming cause it's not really a Button here, more like an accordion.
- Other option is to pass id like we are doing and derive the relationship within the CollapsibleSection component, guessing that's what's currently going on.
- Or have lodash create a unique id for it?

Reply:
The button wraps the collapsible contents in a div with that ID, so I think that's effectively option 2 you have listed?

Reply:
yeah thats what i was guessing.... okay suppose that's fine! but yeah what do we think about `<CollapsibleSection />` or something of that naming?

Reply:
Yeah I like that - maybe just `<Collapsible>`

Reply:
yeah a bit abstract, but cool with it

Reply:
`Section` makes me think `<section>` haha

Reply:
Opted for `CollapsibleContent`!